dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+        float,
+        Vector,
+        Vector,
+    ]:
+        """
+        Get the simulation discharges in batch mode (no user interaction).
+
+        Arguments
+        ---------
+        rivers : RiversObject
+            A dictionary containing the river data.
+        ibranch : int
+            Number of selected branch.
+        ireach : int
+            Number of selected reach.
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        q_stagnant : float
+            Discharge below which the river flow is negligible [m3/s].
+        celerity_hg : float
+            Bed celerity during transitional and flood periods (from rivers configuration file) [m/s].
+        celerity_lw : float
+            Bed celerity during low flow period (from rivers configuration file) [m/s].
+        nwidth : float
+            Normal river width (from rivers configuration file) [m].
+
+        Results
+        -------
+        q_threshold : Optional[float]
+            River discharge at which the measure becomes active [m3/s].
+        three_characteristic_discharges : QRuns
+            Tuple of (at most) three characteristic discharges [m3/s].
+        apply_q : Tuple[bool, bool, bool]
+            A list of 3 flags indicating whether each value should be used or not.
+            The Q1 value can't be set to None because it's needed for char_times.
+        tstag : float
+            Fraction of year during which flow velocity is considered negligible [-].
+        fractions_of_the_year : Vector
+            A vector of values each representing the fraction of the year during which the discharge is given by the corresponding entry in Q [-].
+        rsigma : Vector
+            A vector of values each representing the relaxation factor for the period given by the corresponding entry in Q [-].
+        """
+        q_fit = reach.qfit
+        q_levels = reach.qlevels
+        dq = reach.dq
+
+        q_threshold = self._get_q_threshold_from_config(config)
+        q_bankfull = self._get_q_bankfull_from_config(config, q_threshold, q_levels)
+
+        three_characteristic_discharges, apply_q = dfastmi.kernel.core.char_discharges(q_levels, dq, q_threshold, q_bankfull)
+
+        tstag, fractions_of_the_year, rsigma = dfastmi.kernel.core.char_times(
+            q_fit, q_stagnant, three_characteristic_discharges, celerity_hg, celerity_lw, nwidth
+        )
+
+        q_list = self._discharge_from_config(config, three_characteristic_discharges, apply_q)
+        three_characteristic_discharges = (q_list[0], q_list[1], q_list[2])
+
+        return (
+            q_threshold,
+            three_characteristic_discharges,
+            apply_q,
+            tstag,
+            fractions_of_the_year,
+            rsigma,
+        )
+
+    def _discharge_from_config(self, config : configparser.ConfigParser, three_characteristic_discharges : QRuns, apply_q : BoolVector) -> List:
+        """
```

Comment:
this is a private method, do you want to keep the documentation for it, if yes, please update it.